adapters/repos/db/index.go

```diff
@@ -123,6 +125,23 @@ func NewIndex(ctx context.Context, config IndexConfig,
 
 		shard, err := NewShard(ctx, promMetrics, shardName, index, class)
 		if err != nil {
+			// replace combined error message with one more user friendly in case it refers
+			// to unsupported newly introduced RoaringSet bucket's strategy
+			var errns lsmkv.ErrorNotSupported
+			if errors.As(err, &errns) && errns.Strategy() == lsmkv.SegmentStrategyRoaringSet {
+				r := regexp.MustCompile("creating bucket '(?P<bucketName>.*?)'")
```

Comment:
I would move the regex outside the function into the file scope. 
This way we make sure the check is done only once at application start and the that the application will not crash if someone change the regex to an invalid one.

Having said I don't think we need to parse the bucket name from the error:

1. The bucket name might not make sense to the user since it is internal to the application
2. The error might or might not contain the bucket name.

Therefore I would not parse the bucket name and If I were to track the bucket name I would try to make it part of the error definition